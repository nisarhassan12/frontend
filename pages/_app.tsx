import React from 'react'
import Meta from '../components/Meta'
import App, { Container } from 'next/app'
import { ApolloProvider } from 'react-apollo'
import { ApolloClient } from 'apollo-boost'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import withApollo from '../lib/apollo'
import theme from '../lib/theme'

declare interface Props {
  apollo: ApolloClient<any>
}

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0;
    font-family: ${theme.font}, sans-serif;
  }
  html, body, #__next {
    height: 100%
  }
`

class BaseApp extends App<Props> {
  public render() {
    const { Component, pageProps, apollo } = this.props

    return (
      <Container>
        <GlobalStyle />
        <Meta />
        <ApolloProvider client={apollo}>
          <ThemeProvider theme={theme}>
            <Component {...pageProps} />
          </ThemeProvider>
        </ApolloProvider>
      </Container>
    )
  }
}

export default withApollo(BaseApp)
