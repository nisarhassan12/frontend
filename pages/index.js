import React from 'react'

export default () => <h1>Let's go</h1>