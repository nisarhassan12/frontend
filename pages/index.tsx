import React, { Fragment } from 'react'
import styled from 'styled-components'
// import { Query, QueryResult } from 'react-apollo'
// import { gql, DocumentNode } from 'apollo-boost'
import NavBar from '../components/Layout/NavBar'
import { Page } from '../components/Primitives'
import { Heading } from 'rebass/styled-components'

/* const Status = styled.p`
  width: 40vmax;
  font-size: calc(100% + 0.5vw);
` */

const Links = styled.div`
  a {
    margin-right: 0.7em;
  }
`

const links = [
  {
    href: 'https://github.com/komfy',
    text: 'GitHub 🐙'
  },
  {
    href: 'https://twitter.com/KomfySocial',
    text: 'Twitter 🐦'
  },
  {
    href: 'https://codecov.io/gh/komfy/frontend',
    text: 'CodeCov 📊'
  },
  {
    href: 'https://travis-ci.org/komfy/frontend',
    text: 'Travis CI ⛓'
  },
  {
    href: 'https://t.me/komfy',
    text: 'Telegram 📰'
  }
]

const Index = () => {
  return (
    <Fragment>
      <NavBar />
      <Page flexDirection="column" alignItems="center" justifyContent="center">
        <Heading as="h1" fontSize={['4em', '6em', '8em']}>
          Komfy
        </Heading>

        <h2>Project Status ⚡</h2>
        <p>
          Komfy is a protected social network without annoying trackers and context ads. It is currently in active
          development. User registration will be available soon.
        </p>
        <Links>
          {links.map((link: { href: string; text: string }, i: number) => (
            <a href={link.href} key={i}>
              {link.text}
            </a>
          ))}
        </Links>
        <p>
          I&apos;ll answer all questions here:&nbsp;
          <a href="https://t.me/talentless_guy">@talentless_guy</a>
        </p>
        {/* <Query query={HELLO_WORLD}>
          {({ loading, error, data }: QueryResult) => {
            if (error) return <h1>{error}</h1>
            if (loading || !data) return <div>Loading</div>

            return <div>{JSON.stringify(data)}</div>
          }}
        </Query> */}
      </Page>
    </Fragment>
  )
}
export default Index
