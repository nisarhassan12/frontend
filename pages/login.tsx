import React, { Fragment, useState } from 'react'
import { Button } from 'rebass/styled-components'
import { BasicContainer, Page } from '../components/Primitives'
import NavBar from '../components/Layout/NavBar'
import Form from '../components/Layout/Form'

const Login = () => {
  const [name, setName] = useState<string>('')

  return (
    <Fragment>
      <NavBar />
      <Page alignItems="center" justifyContent="center" flexDirection="column">
        <h1>Welcome back, {name}!</h1>
        <BasicContainer>
          <Form
            onSubmit={() => {}}
            body={[
              {
                slug: 'Username',
                name: 'username',
                type: 'text',
                required: true,
                onInput: e => setName(e.currentTarget.value),
                placeholder: 'How were you called?'
              },
              {
                slug: 'Password',
                name: 'password',
                placeholder: 'What was your password?',
                required: true
              }
            ]}>
            <Button variant="secondary" type="submit">
              Sign In
            </Button>
          </Form>
        </BasicContainer>
      </Page>
    </Fragment>
  )
}

export default Login
