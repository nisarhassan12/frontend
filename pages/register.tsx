import React, { useState, useEffect } from 'react'
import { Heading as H1, Card, Flex } from 'rebass'
import styled from 'styled-components'
import NavBar from '../components/Layout/NavBar'
import Form from '../components/Layout/Form'

const Heading = styled(H1)`
  font-size: ${({ theme }) => theme.h1};
`

const Register = () => {
  const [name, setName] = useState<string>('')
  const [pwd, setPwd] = useState<string>('')
  const [pwdLevel, setPwdLevel] = useState<string>('off')

  useEffect(() => {
    fetch(`https://api.komfy.now.sh/rand${pwdLevel === 'on' && '_dict'}`)
      .then(res => res.text())
      .then(text => setPwd(text))
  }, [pwd])

  return (
    <Flex alignItems="center" justifyContent="center" flexDirection="column" flexWrap="wrap" css={{ height: '100vh' }}>
      <NavBar />
      <Heading as="h1" textAlign="center">
        Hello {name}!
      </Heading>
      <Card boxShadow="0 0 16px rgba(0, 0, 0, .25)" m="2em" p="2em">
        <Form
          to="/"
          action="https://api.komfy.now.sh/reg"
          body={[
            {
              name: 'username',
              type: 'text',
              onInput: e => setName(e.currentTarget.value),
              placeholder: 'How we can call you?'
            },
            {
              name: 'email',
              type: 'email',
              placeholder: 'hello@example.com'
            }
          ]}
          method="POST">
          <button onClick={() => setPwd(pwd)}>Create password</button>
          <div>
            Complex password:
            <input type="checkbox" value={pwdLevel} onChange={() => setPwdLevel(pwdLevel ? 'on' : 'off')} />
          </div>

          {pwd}
          <input name="password" value={pwd} disabled />
        </Form>
      </Card>
    </Flex>
  )
}

export default Register
