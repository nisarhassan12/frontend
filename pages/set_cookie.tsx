import React from 'react'
import Router from 'next/router'
import cookies from 'nookies'

const SetCookie = () => <h1>Redirecting to Sign In...</h1>

SetCookie.getInitialProps = async (ctx: any) => {
  const { req, res, query } = ctx

  if (!query.verify_code) {
    res.statusCode = 400
    res.end('Verify code not found')
  } else {
    const verify_code = query.verify_code.toString()

    cookies.set({ req, res }, 'jwt-token', verify_code, {})
    if (res) {
      res.writeHead(302, {
        Location: '/login'
      })
      res.end()
    } else {
      Router.push('/login')
    }
  }

  return {}
}

export default SetCookie
