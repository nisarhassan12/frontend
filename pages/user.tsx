import * as React from 'react'
import 'isomorphic-unfetch'
import { NextContext } from 'next'
import '../styles/main.css'

type Data = {
    user: string
}

const UserPage = (props: any) => (
    <div>Hello, {props.user}!</div>
)

UserPage.getInitialProps = async ({ query }: NextContext<Data>) => {
    return {
        user: query.user
    }
}

export default UserPage