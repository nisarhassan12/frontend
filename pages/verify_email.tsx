import React, { Fragment } from 'react'
import { Flex } from 'rebass'
import { BasicContainer } from '../components/Primitives'
import NavBar from '../components/Layout/NavBar'

const VerifyEmail = () => (
  <Fragment>
    <NavBar />
    <Flex style={{ height: '80vh' }} alignItems="center" justifyContent="center">
      <BasicContainer>
        <h1>Check your email</h1>
      </BasicContainer>
    </Flex>
  </Fragment>
)

export default VerifyEmail
