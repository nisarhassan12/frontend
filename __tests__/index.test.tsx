import React from 'react'
import { create } from 'react-test-renderer'
import { Index } from '../pages/index'
import User from '../pages/user'
import { ApolloProvider } from '@apollo/react-hooks'
import { ApolloClient } from 'apollo-client'
import { HttpLink, InMemoryCache } from 'apollo-boost'

it('Renders user page correctly', () => {
  const tree = create(<User id="Ivan" />).toJSON()
  expect(tree).toMatchSnapshot()
})

it(`Doesn't fail on receiving Apollo data`, () => {
  const tree = create(
    <ApolloProvider
      client={
        new ApolloClient({
          link: new HttpLink({ uri: 'https://api.komfy.now.sh' }),
          cache: new InMemoryCache()
        })
      }>
      <Index />
    </ApolloProvider>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

it('Shows error text when wrong link', () => {
  const tree = create(
    <ApolloProvider
      client={
        new ApolloClient({
          link: new HttpLink({ uri: 'https://example.com' }),
          cache: new InMemoryCache()
        })
      }>
      <Index />
    </ApolloProvider>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
