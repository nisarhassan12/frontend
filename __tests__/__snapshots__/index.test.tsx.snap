// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doesn't fail on receiving Apollo data 1`] = `
<div
  className="pages__Page-sc-9mrxf5-0 jnmkXX"
>
  <h1
    className="pages__Header-sc-9mrxf5-1 SMwOW"
  >
    Komfy
  </h1>
  <h2>
    Project Status ⚡
  </h2>
  <div
    className="pages__Links-sc-9mrxf5-3 fwJWiw"
  >
    <a
      href="https://github.com/komfy"
      onClick={[Function]}
    >
      GitHub
    </a>
    <a
      href="https://crowdforge.io/projects/534"
      onClick={[Function]}
    >
      CrowdForge
    </a>
    <a
      href="https://twitter.com/KomfySocial"
      onClick={[Function]}
    >
      Twitter
    </a>
  </div>
  <p
    className="pages__Status-sc-9mrxf5-2 hGXJBu"
  >
    Soon there will be new stuff. You'll be able to create an account and publish images in text. Pre-alpha is close.
  </p>
  <p>
    I'll answer all questions here: 
    <a
      href="https://t.me/talentless_guy"
      onClick={[Function]}
    >
      @talentless_guy
    </a>
  </p>
  <div>
    <span>
      GraphQL request:
      <b>
         
      </b>
    </span>
  </div>
  <a
    href="/user/4219319"
    onClick={[Function]}
  >
    Random user page
  </a>
  <a
    href="/error"
    onClick={[Function]}
  >
    Example error page
  </a>
</div>
`;

exports[`Renders user page correctly 1`] = `
<div
  className="pages__Page-sc-9mrxf5-0 jnmkXX"
>
  <p>
    Hello, #
    Ivan
    !
  </p>
  <p>
    There will be content soon.
  </p>
  <p>
    You need to wait a bit until we release our pre-alpha.
  </p>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;

exports[`Shows error text when wrong link 1`] = `
<div
  className="pages__Page-sc-9mrxf5-0 jnmkXX"
>
  <h1
    className="pages__Header-sc-9mrxf5-1 SMwOW"
  >
    Komfy
  </h1>
  <h2>
    Project Status ⚡
  </h2>
  <div
    className="pages__Links-sc-9mrxf5-3 fwJWiw"
  >
    <a
      href="https://github.com/komfy"
      onClick={[Function]}
    >
      GitHub
    </a>
    <a
      href="https://crowdforge.io/projects/534"
      onClick={[Function]}
    >
      CrowdForge
    </a>
    <a
      href="https://twitter.com/KomfySocial"
      onClick={[Function]}
    >
      Twitter
    </a>
  </div>
  <p
    className="pages__Status-sc-9mrxf5-2 hGXJBu"
  >
    Soon there will be new stuff. You'll be able to create an account and publish images in text. Pre-alpha is close.
  </p>
  <p>
    I'll answer all questions here: 
    <a
      href="https://t.me/talentless_guy"
      onClick={[Function]}
    >
      @talentless_guy
    </a>
  </p>
  <div>
    <span>
      GraphQL request:
      <b>
         
      </b>
    </span>
  </div>
  <a
    href="/user/3837942"
    onClick={[Function]}
  >
    Random user page
  </a>
  <a
    href="/error"
    onClick={[Function]}
  >
    Example error page
  </a>
</div>
`;
